configuration/src/main/java/io/camunda/configuration/beanoverrides/BrokerBasedPropertiesOverride.java

```diff
@@ -809,111 +796,21 @@ private void populateCamundaExporter(final BrokerBasedProperties override) {
-      setArg(args, "notifier.auth0Protocol", database.getIncidentNotifier().getAuth0Protocol());
-      setArg(args, "notifier.m2mClientId", database.getIncidentNotifier().getM2mClientId());
-      setArg(args, "notifier.m2mClientSecret", database.getIncidentNotifier().getM2mClientSecret());
-      setArg(args, "notifier.m2mAudience", database.getIncidentNotifier().getM2mAudience());
-    }
-
-    setArg(
-        args, "batchOperationCache.maxCacheSize", database.getBatchOperationCache().getMaxSize());
-    setArg(args, "processCache.maxCacheSize", database.getProcessCache().getMaxSize());
-    setArg(
-        args,
-        "decisionRequirementsCache.maxCacheSize",
-        database.getDecisionRequirementsCache().getMaxSize());
-    setArg(args, "formCache.maxCacheSize", database.getFormCache().getMaxSize());
-
-    setArg(args, "postExport.batchSize", database.getPostExport().getBatchSize());
-    // Duration supports only seconds and nanos; other units need to be converted
-    setArg(
-        args,
-        "postExport.delayBetweenRuns",
-        database.getPostExport().getDelayBetweenRuns().getSeconds() * 1000);
-    setArg(
-        args,
-        "postExport.maxDelayBetweenRuns",
-        database.getPostExport().getMaxDelayBetweenRuns().getSeconds() * 1000);
-    setArg(args, "postExport.ignoreMissingData", database.getPostExport().isIgnoreMissingData());
-    setArg(
-        args,
-        "batchOperation.exportItemsOnCreation",
-        database.getBatchOperations().isExportItemsOnCreation());
-
-    setArg(args, "bulk.delay", database.getBulk().getDelay().getSeconds());
-    setArg(args, "bulk.size", database.getBulk().getSize());
-    setArg(args, "bulk.memoryLimit", database.getBulk().getMemoryLimit().toMegabytes());
-
-    final var auditLog = unifiedConfiguration.getCamunda().getData().getAuditLog();
     exporter.setArgs(
         ExporterConfiguration.of(io.camunda.exporter.config.ExporterConfiguration.class, args)
-            .apply(config -> config.setAuditLog(auditLog.toConfiguration()))
+            .apply(
+                config -> {
+                  CamundaExporterConfigurationApplier.applyRetention(config, unifiedConfiguration);
+                  CamundaExporterConfigurationApplier.applyConnect(config, unifiedConfiguration);
+                  CamundaExporterConfigurationApplier.applyIndex(config, unifiedConfiguration);
+                  CamundaExporterConfigurationApplier.applyHistory(config, unifiedConfiguration);
+                  CamundaExporterConfigurationApplier.applyPostExportConfiguration(
+                      config, unifiedConfiguration);
+                  CamundaExporterConfigurationApplier.applyBulk(config, unifiedConfiguration);
+                  CamundaExporterConfigurationApplier.applyIncidentNotifier(
+                      config, unifiedConfiguration);
+                  CamundaExporterConfigurationApplier.applyMisc(config, unifiedConfiguration);
+                })
```

Comment:
🫶 nice (we could use static imports to make this a bit less chatty, but that's a super nitpick - looks good to me)